### PR TITLE
Draft: Development PR

### PR DESCRIPTION
**Pull request by Hyperb0rean**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,27 +1,25 @@
...
 
 RUN apt-get update && \
     apt-get install -y \
     libboost-dev libboost-system-dev \
     cmake 
 
+
 RUN apt-get install g++ make binutils libboost-system-dev libssl-dev zlib1g-dev libcurl4-openssl-dev
 
+#installing tgbot lib
```
Comment (vityaman) [inline Dockerfile:13]: Maybe add a space between comment token and text to improve readability.

File: src/main.cpp [hunk 2]
```diff
@@ -16,18 +22,84 @@ class Command : public TgBot::BotCommand {
     }
 };
 
+bool HandleUser(size_t id, std::unordered_map<size_t, size_t>& users) {
```
Comment (vityaman) [inline src/main.cpp]: Not so obvious that `id` is a user id, a have been thinking for a 3 seconds to got it.

`users` I guess is an amount of messages be user id.

It is better to give it a more detailed name.

File: src/main.cpp [hunk 3]
```diff
@@ -16,18 +22,84 @@ class Command : public TgBot::BotCommand {
...
+    return users[id] > kMaxQueries;
+}
+
+std::vector<std::string> GetArguments(std::string message) {
+    std::vector<std::string> tokens;
+    boost::split(tokens, message, boost::is_any_of(" "));
+    return tokens;
+}
+
+std::vector<std::string> ReadDDLQueries(std::string filename) {
```
Comment (vityaman) [inline src/main.cpp]: Logic can be extracted to function that takes an iostream for parsing, and this function just simply open the file and call that function.

File: src/main.cpp [hunk 4]
```diff
@@ -38,16 +110,23 @@ int main() {
         bot.getApi().sendMessage(message->chat->id, "Unknown command: " + message->text);
     });
 
-    bot.getEvents().onCommand("start", [&bot](TgBot::Message::Ptr message) {
+    bot.getEvents().onCommand("start", [&bot, &users](TgBot::Message::Ptr message) {
+        if (HandleUser(message->chat->id, users)) {
```
Comment (vityaman) [inline src/main.cpp]: Confusing as I expected that this function is void by its name.

File: src/main.cpp [hunk 5]
```diff
@@ -68,20 +151,69 @@ int main() {
         }
         bot.getApi().sendMessage(message->chat->id, std::move(result));
     });
+    bot.getEvents().onCommand(
+        "get_friend_email", [&bot, &bot_commands, &users](TgBot::Message::Ptr message) {
```
Comment (vityaman) [inline src/main.cpp]: Name of a command is duplicated in a map and here.

File: src/main.cpp [hunk 6]
```diff
@@ -68,20 +151,69 @@ int main() {
...
+            }
+            std::string result;
+            auto args = GetArguments(message->text);
+
+            pqxx::connection conn;
+            pqxx::work query(conn);
+            try {
+
+                auto resp = std::format(
+                    "insert into crops (name, humidity, brightness, is_legal, "
```
Comment (vityaman) [inline src/main.cpp]: Sql injection?

File: src/main.cpp [hunk 7]
```diff
@@ -68,20 +151,69 @@ int main() {
...
         TgBot::TgLongPoll longPoll(bot);
         while (true) {
-            std::cerr << GetCurrentTime() << " | Long poll started\n";
+            std::cerr << GetCurrentTime() << " | Long poll started on thread " << 0 << "\n";
             longPoll.start();
             if (!is_alive) {
+                for (auto& [id, count] : users) {
+                    bot.getApi().sendMessage(id, "Guys, I've been killed(");
+                }
                 std::exit(0);
```
Comment (vityaman) [inline src/main.cpp]: Why not just return 0?

File: Dockerfile (hunk 1 above)
Comment (Hyperb0rean) [reply Dockerfile:13]: fixed

File: src/main.cpp (hunk 3 above)
Comment (Hyperb0rean) [reply src/main.cpp]: done

File: src/main.cpp (hunk 2 above)
Comment (Hyperb0rean) [reply src/main.cpp]: done

File: src/main.cpp (hunk 4 above)
Comment (Hyperb0rean) [reply src/main.cpp]: renamed to CheckDDOSUser

File: src/main.cpp (hunk 6 above)
Comment (Hyperb0rean) [reply src/main.cpp]: rewritten to prepared_statements

File: src/main.cpp (hunk 7 above)
Comment (Hyperb0rean) [reply src/main.cpp]: yep

File: src/main.cpp (hunk 5 above)
Comment (Hyperb0rean) [reply src/main.cpp]: rewritten a bit

**Pull request merged**